modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/parameter/ObjectTypeParameterParser.java

```diff
@@ -85,13 +88,22 @@ protected void doParse(Builder definitionBuilder) throws ConfigurationException
     type.getFields().forEach(this::parseField);
   }
 
+  private String getFieldName(ObjectFieldType objectFieldType, String fieldName) {
+
+    return objectFieldType.getValue().getAnnotation(TypeIdAnnotation.class)
+        .map(typeIdAnnotation -> typeIdAnnotation.getValue())
+        .map(typeIdName -> InfrastructureTypeMapping.getNameMap().get(typeIdName))
+        .filter(infrastructureName -> !StringUtils.isBlank(infrastructureName)).orElse(fieldName);
+  }
+
   private void parseField(ObjectFieldType objectField) {
     final MetadataType fieldType = objectField.getValue();
-    final String fieldName = objectField.getKey().getName().getLocalPart();
+    final String key = objectField.getKey().getName().getLocalPart();
+    final String fieldName = getFieldName(objectField, key);
     final boolean acceptsReferences = acceptsReferences(objectField);
     final Object defaultValue = getDefaultValue(fieldType).orElse(null);
     final ExpressionSupport expressionSupport = getExpressionSupport(objectField);
-    Optional<DslElementSyntax> fieldDsl = typeDsl.getChild(fieldName);
+    Optional<DslElementSyntax> fieldDsl = typeDsl.getChild(key);
```

Comment:
I'm not really sure about doing that. An infrastructure parameter is kinda special, but it is still a parameter. You are going to have a different key name but the parsing is all the same. I know that we can shortcut it and go straight to the `parseObject(...)` thing, but if the parameter parsing changes in the future, we are still going to want those infrastructure parameters to be parsed in the same way as the others.